tool/gocross/toolchain.go

```diff
@@ -60,7 +60,12 @@ func getToolchain() (toolchainDir, gorootDir string, err error) {
 		return "", "", err
 	}
 
-	cache := filepath.Join(os.Getenv("HOME"), ".cache")
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", "", err
+	}
+
+	cache := filepath.Join(homeDir, ".cache")
```

Comment:
maybe leave a comment that we're explicitly using ".cache" everywhere instead of `os.UserCacheDir` for legacy reasons (because we historically did so) and we don't want to break old macOS users who already had their ~/.cache populated?